src/window.c

```diff
@@ -1393,6 +1411,64 @@ static void draw_number_box(struct world *mzx_world, struct dialog *di,
   }
 }
 
+static int get_slot_state(int slot, boolean *highlighted_slots,
+ boolean *disabled_slots)
+{
+  if(disabled_slots != NULL && disabled_slots[slot])
+    return SLOT_DISABLED;
+
+  if(highlighted_slots != NULL && highlighted_slots[slot])
+    return SLOT_HIGHLIGHT;
+
+  return SLOT_NORMAL;
+}
+
+static void draw_slot_selector(struct world *mzx_world, struct dialog *di,
+ struct element *e, int color, int active)
+{
+  struct slot_selector *src = (struct slot_selector *)e;
+  int x = di->x + e->x;
+  int y = di->y + e->y;
+  int slot_x;
+  int slot_y = y + SLOTSEL_BAR_Y;
+  int i;
+  int slot_color;
+
+  write_string(src->title, x, y, color, 0);
+
+  for(i = 0; i < src->num_slots; i++)
+  {
+    slot_x = x + (i * 4);
+    switch(get_slot_state(i, src->highlighted_slots, src->disabled_slots))
+    {
+      case SLOT_DISABLED:
+      {
+        slot_color = DI_SLOTSEL_DISABLE;
+        break;
+      }
+
+      case SLOT_HIGHLIGHT:
+      {
+        slot_color = i == src->selected_slot
+         ? DI_ACTIVEBUTTON
+         : DI_SLOTSEL_HIGHLIGHT;
+        break;
+      }
+
+      case SLOT_NORMAL:
+      {
+        slot_color = i == src->selected_slot
+         ? DI_ARROWBUTTON
+         : DI_SLOTSEL_NORMAL;
+        break;
+      }
+    }
+
+    fill_line(4, slot_x, slot_y, 0, slot_color);
+    write_number(i + 1, slot_color, slot_x + 2, slot_y, 1, true, 10);
```

Comment:
(Pedantry.) These lines emit a warning with both devkitARM r53 (NDS only) and devkitPSP r16:
```src/window.c: In function 'draw_slot_selector':
src/window.c:1468:5: warning: 'slot_color' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1468 |     fill_line(4, slot_x, slot_y, 0, slot_color);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
```
src/window.c: In function 'draw_slot_selector':
src/window.c:1468:14: warning: 'slot_color' may be used uninitialized in this function [-Wuninitialized]
```

Reply:
This should be fixed in 49010f1.